lib/purgatory/purgatory.rb

```diff
@@ -47,6 +47,7 @@ def approve!(approver = nil)
     return false if approved? 
     if soul_with_changes.save
       soul.send(performable_method[:method],*performable_method[:args]) if performable_method
+      return false if soul.errors.any?
```

Comment:
As per our discussion I think we should do the following:
a) Either call the performable method if it exists, or call `soul_with_changes.save` if it doesn't. No need to do both.
b) The result of whatever we do in a) should be a boolean and that can drive the logic, instead of checking for errors.